### PR TITLE
Update Ruby to `2.7.4` and other gem dependencies

### PR DESCRIPTION
**Pull request by waldofouche**
Updates Ruby and gem dependencies:

- Ruby from 2.7.3 to 2.7.4
- All other gem dependencies to latest versions

This is part of https://github.com/ausaccessfed/stockpile/issues/87

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -317,34 +324,34 @@ GEM
...
       railties (>= 3.2)
       torba (~> 1.0)
     tzinfo (1.2.9)
       thread_safe (~> 0.1)
     uglifier (4.2.0)
       execjs (>= 0.3.0, < 3)
-    unicode-display_width (1.7.0)
-    unicorn (5.7.0)
+    unicode-display_width (2.0.0)
+    unicorn (6.0.0)
```
Comment (rianniello) [inline Gemfile.lock:341]: Have to watch this one carefully when rolling out, could be breaking.

Comment (rianniello) [reply Gemfile.lock:341]: (unicorn 6)

Comment (rianniello) [reply Gemfile.lock:341]: Though it looks like `puma` is in here too 🤔 Odd.

Comment (rianniello) [reply Gemfile.lock:341]: https://github.com/ausaccessfed/reporting-service/blob/develop/config/god.rb#L34

**Review by rianniello — APPROVED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (rianniello) [reply Gemfile.lock:341]: (unicorn used)

Comment (waldofouche) [reply Gemfile.lock:341]: Okay I will keep an eye on this one when it has been rolled out

**Pull request merged**